File: src/main/java/dev/vrba/slack/tenor/bot/service/tenor/TenorApiClient.java
package dev.vrba.slack.tenor.bot.service.tenor;

import dev.vrba.slack.tenor.bot.service.tenor.dto.TenorSearchResponse;
import io.micronaut.core.annotation.NonNull;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.QueryValue;
import io.micronaut.http.client.annotation.Client;
import reactor.core.publisher.Mono;

@Client(id = "tenor")
public interface TenorApiClient {

    @Get("/v2/search")
    @NonNull
    Mono<TenorSearchResponse> search(
        @NonNull @QueryValue("q") String query,
        @NonNull @QueryValue("key") String key
    );
}

File: src/main/kotlin/dev/vrba/slack/service/tenor/TenorApiClient.kt
package dev.vrba.slack.service.tenor

import dev.vrba.slack.service.tenor.dto.TenorSearchResponse
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.QueryValue
import io.micronaut.http.client.annotation.Client

@Client(id = "tenor")
interface TenorApiClient {
    @Get("/v2/search")
    suspend fun search(
        @QueryValue("q") q: String,
        @QueryValue("limit") limit: Int = 20,
        @QueryValue("key") key: String = "",
    ): TenorSearchResponse
}
